### DISCUSSION TITLE
`rbenv install 3.0.4` fails on macOS 13.3.1

### DISCUSSION THREAD

**Discussion by coolandsmartrr** [rbenv/ruby-build#2185] [Build failures] [answered]
I've tried to install ruby 3.0.4 through rbenv with the command: `rbenv install 3.0.4 `.
I get an error relating OpenSSL all the time:

```
The Ruby openssl extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Configure options used:
  --prefix=/Users/ricky/.rbenv/versions/3.0.4
  --enable-shared
  --with-readline-dir=/opt/homebrew/opt/readline
  --with-libyaml-dir=/opt/homebrew/opt/libyaml
  --with-openssl-dir=/usr/local/opt/openssl@1.1
  CC=clang
  LDFLAGS=-L/Users/ricky/.rbenv/versions/3.0.4/lib 
  CPPFLAGS=-I/Users/ricky/.rbenv/versions/3.0.4/include 
```

To fix the OpenSSL error, I've tried these methods:

1. Reinstall OpenSSL via `brew uninstall openssl` and  `brew install openssl`, then `rbenv install 3.0.4`
2. Make sure OpenSSL is linked via `brew link openssl --force`, then `rbenv install 3.0.4`
3. Install with OpenSSL path specified: `RUBY_CONFIGURE_OPTS=--with-openssl-dir=/opt/homebrew/opt/openssl@3 rbenv install 3.0.4`
4. Reinstall rbenv with `brew install rbenv ruby-build`, then `rbenv install 3.0.4`

These methods were indicated in this [Github dicussion](https://github.com/rbenv/ruby-build/issues/377). None of these methods resulted in a successful build.

Please let me know if there is a way to build ruby successfully.

Here is my configuration:

> 
> OS: macOS 13.3.1
> Chip: Apple Silicon
> rbenv: 1.2.0
> Homebrew: 4.0.12
>

**Comment by hsbt** [comment]
Ruby 3.0 didn't support OpenSSL 3.x. Try `brew install openssl@1.1` and replacing 
 openssl version on `RUBY_CONFIGURE_OPTS`.

**Reply by coolandsmartrr** [reply]
@hsbt Thank you for your reply. I got this error:

```
The Ruby openssl extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Configure options used:
  --prefix=/Users/ricky/.rbenv/versions/3.0.4
  --enable-shared
  --with-readline-dir=/opt/homebrew/opt/readline
  --with-libyaml-dir=/opt/homebrew/opt/libyaml
  --with-openssl-dir=/opt/homebrew/opt/openssl@1.1
  CC=clang
  LDFLAGS=-L/Users/ricky/.rbenv/versions/3.0.4/lib 
  CPPFLAGS=-I/Users/ricky/.rbenv/versions/3.0.4/include 
```

I tried using openssl@1.1 with these steps:
1. Take path printed from `brew --prefix openssl@1.1`
2. Run `RUBY_CONFIGURE_OPTS=--with-openssl-dir=/opt/homebrew/opt/openssl@1.1  rbenv install 3.0.4`

**Reply by coolandsmartrr** [reply]
I ran these exports and rbenv built ruby successfully:

```
export LDFLAGS=-L/opt/homebrew/opt/openssl@1.1/lib
export CPPFLAGS=-I/opt/homebrew/opt/openssl@1.1/include
```

**Reply by hsbt** [reply]
You may need to run with double quote for `RUBY_CONFIGURE_OPTS`.

```
RUBY_CONFIGURE_OPTS="--with-openssl-dir=/opt/homebrew/opt/openssl@1.1" rbenv install 3.0.4
```

----

I'm happy to your issue was solved. Thanks for sharing that.

**Reply by coolandsmartrr** [reply]
Thank you for helping me out.

**Comment by joekrump** [comment]
I'm running into almost the identical issue for installing ruby 3.0.6 and the answer above doesn't resolve my issue. 

When I run:
```
RUBY_CONFIGURE_OPTS="--with-openssl-dir=/opt/homebrew/opt/openssl@1.1" rbenv install 3.0.6
```

I get:

```
The Ruby openssl extension was not compiled.
/Users/josephkrump/.rbenv/versions/3.0.6/lib/ruby/3.0.0/yaml.rb:3: warning: It seems your ruby installation is missing psych (for YAML output).
To eliminate this warning, please install libyaml and reinstall your ruby.
The Ruby yaml extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Configure options used:
  --prefix=/Users/josephkrump/.rbenv/versions/3.0.6
  --enable-shared
  --with-readline-dir=/opt/homebrew/opt/readline
  --with-libyaml-dir=/opt/homebrew/opt/libyaml
  --with-gmp-dir=/opt/homebrew/opt/gmp
  --with-openssl-dir=/opt/homebrew/opt/openssl@1.1
  CC=clang
  LDFLAGS=-L/Users/josephkrump/.rbenv/versions/3.0.6/lib -L/opt/homebrew/opt/openssl@1.1/lib
  CPPFLAGS=-I/Users/josephkrump/.rbenv/versions/3.0.6/include -I/opt/homebrew/opt/openssl@1.1/include
```

>OS: macOS 13.0
Chip: Apple Silicon
rbenv: 1.2.0
Homebrew: 4.0.12

I followed the steps here: https://github.com/rbenv/ruby-build/wiki#macos to get setup.

Any help is appreciated.

**Reply by hsbt** [reply]
>The Ruby yaml extension was not compiled.

You should install `libyaml` via homebrew or macports.

**Reply by joekrump** [reply]
I do have it installed which is also a confusing part. When I run `rbenv install 3.0.6` I can see that in the output:
```
To follow progress, use 'tail -f /var/folders/wq/kmqzgfsx6lz0fwj35x2w83ww0000gn/T/ruby-build.20230412203424.56165.log' or pass --verbose
Downloading openssl-1.1.1t.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/8dee9b24bdb1dcbf0c3d1e9b02fb8f6bf22165e807f45adeb7c9677536859d3b
Installing openssl-1.1.1t...
Installed openssl-1.1.1t to /Users/josephkrump/.rbenv/versions/3.0.6

Downloading ruby-3.0.6.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.6.tar.gz
Installing ruby-3.0.6...
ruby-build: using readline from homebrew
ruby-build: using libyaml from homebrew
ruby-build: using gmp from homebrew
```

**Reply by joekrump** [reply]
One more thing that is odd is that if I run `rbenv install 3.0.5` it installs successfully without issue.

**Reply by hsbt** [reply]
Interesting. There is no changes related `Psych` and `libyaml` at Ruby 3.0.6. see https://github.com/ruby/ruby/compare/v3_0_5...v3_0_6

**Reply by joekrump** [reply]
So I was able to get 3.0.6 installed by first calling `unset GEM_HOME` which I'd forget had been configured as a part of a recent Flutter setup that I'd done where I'd added:
```
export GEM_HOME=$HOME/.gem
export PATH=$GEM_HOME/bin:$PATH
```
to my `.zshrc` file. 

I was alerted to the fact that I'd set `GEM_HOME` when I installed `rvm` to try to see if I could successfully install 3.0.6 with that and it warned me that `GEM_HOME` had already been set. 

After running `unset GEM_HOME` I was able to just run `rbenv install 3.0.6` without issue.

**Reply by joekrump** [reply]
Thanks for taking some time to look into this and for your prompt responses @hsbt

**Reply by hsbt** [reply]
👍 Sharing `GEM_HOME` is useful. But it confused us when we used multiple versions of Ruby. I'll resolve this issue at Ruby and RubyGems.

**Reply by mislav** [reply]
@hsbt Should ruby-build delete the GEM_HOME variable inherited from the environment while building Ruby? I'm not sure if the user's GEM_HOME value is useful for anything when building Ruby, and could potentially [sabotage the installation](https://github.com/rbenv/ruby-build/discussions/2170#discussioncomment-5476325) if that location contains gems built by another Ruby version.

**Reply by hsbt** [reply]
>Should ruby-build delete the GEM_HOME variable inherited from the environment while building Ruby?

I think so. But the recent Ruby version will install `debug` or some of useful gems into `GEM_HOME`. When ruby-build delete the `GEM_HOME`, `make install` try to install the default location of Ruby installation.

After that, the users can't find `debug` gem and others. I investigate what happens deleting `GEM_HOME`.

**Reply by erikw** [reply]
I've been having issues with asdf-ruby/rbenv/ruby-build for quite some time, with the ruby installation failing with building openssl extensions, and I just found out it's because I had `$GEM_HOME` set in my environment.

Some other stories here: https://github.com/asdf-vm/asdf-ruby/issues/206#issuecomment-860106503

**Comment by sudeeptarlekar** [comment]
Hello,
First of all thanks for this wonderful tool with which I never had a problem before but recently I installed Ventura(13.3.1) and now I am not able to install any Ruby on my Mac.
I have tried every answer from discussions but nothing worked.

Using XCode14
> $> pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | grep version
   version: 14.3.0.0.1.1679647830

Reinstalled libyaml, libffi, readline, gmp and openssl@1.1

And after that installing Ruby@2.7.7 is failing
>  RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.1) --disable-install-doc --disable-shared" rbenv install 2.7.7

Here are last few lines from log

```
installing default fiddle libraries
linking shared-object rbconfig/sizeof.bundle
compiling ripper.c
compiling init.c
linking shared-object readline.bundle
compiling stringio.c
checking ../.././parse.y and ../.././ext/ripper/eventids2.c
ripper.c:6178:9: warning: variable 'yynerrs' set but not used [-Wunused-but-set-variable]
    int yynerrs;
        ^
ripper.y:10722:1: warning: unused function 'parser_token_value_print' [-Wunused-function]
parser_token_value_print(struct parser_params *p, enum yytokentype type, const YYSTYPE *valp)
^
2 warnings generated.
make[2]: make[2]: *** [init.o] Killed: 9
make[2]: *** [../../.ext/.timestamp/.RUBYCOMMONDIR.-.psych.-.visitors.time] Killed: 9
make[2]: *** [stringio.o] Killed: 9
make[2]: *** [strscan.o] Killed: 9*** [pre-install-rb-default] Killed: 9
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [ext/stringio/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [ext/rubyvm/all] Error 2
make[1]: *** [ext/psych/all] Error 2

make[1]: *** [ext/strscan/all] Error 2
compiling syslog.c
compiling constants.c
linking shared-object ripper.bundle
compiling zlib.c
linking shared-object syslog.bundle
make[1]: *** [ext/socket/all] Error 2
linking shared-object zlib.bundle
make: *** [build-ext] Error 2
```

complete log can be found at https://gist.github.com/sudeeptarlekar/782fe0f289bae491dc8a284bba4f5633

Tried installing Ruby3 with openssl@3, but not able to install that either.

Let me know if you need more information from my end.

**Reply by mislav** [reply]
Hi, your failure doesn't seem at all related to the original post. I'm not sure what causes it, but you could try without `--disable-shared` (there is a known Ruby compilation problem on macOS when the shared library is disabled) or with a newer Ruby version, since 2.7 is EOL.

**Reply by sudeeptarlekar** [reply]
Thank you @mislav for looking into it, but I am able to install Ruby now, looks like issue of corporate MacOS image. Few weeks ago I installed few Xcode updates and now able to install Ruby as usual...

**Comment by hellohuaxuan** [comment]
I fix the issue brew link --overwrite openssl@1.1

**Reply by baskint** [reply]
this worked for me as well. Thanks for all the responses.
```
>  brew link --overwrite openssl@1.1
Linking /opt/homebrew/Cellar/openssl@1.1/1.1.1w... 9077 symlinks created.

If you need to have this software first in your PATH instead consider running:
  echo 'export PATH="/opt/homebrew/opt/openssl@1.1/bin:$PATH"' >> ~/.zshrc
  ```
  followed by
  ````
> RUBY_CONFIGURE_OPTS="--with-openssl-dir=/opt/homebrew/opt/openssl@1.1" rbenv install 3.0.4
==> Downloading ruby-3.0.4.tar.gz...
-> curl -q -fL -o ruby-3.0.4.tar.gz https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.4.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 20.1M  100 20.1M    0     0  6817k      0  0:00:03  0:00:03 --:--:-- 6826k
==> Installing ruby-3.0.4...
ruby-build: using readline from homebrew
ruby-build: using libyaml from homebrew
ruby-build: using gmp from homebrew
-> ./configure "--prefix=$HOME/.rbenv/versions/3.0.4" --enable-shared --with-readline-dir=/opt/homebrew/opt/readline --with-libyaml-dir=/opt/homebrew/opt/libyaml --with-gmp-dir=/opt/homebrew/opt/gmp --with-openssl-dir=/opt/homebrew/opt/openssl@1.1
-> make -j 8
-> make install
-> "$HOME/.rbenv/versions/3.0.4/bin/ruby" -e <SCRIPT>
==> Installed ruby-3.0.4 to /Users/baskint/.rbenv/versions/3.0.4

**Reply by nizarhmain** [reply]
Worked for me too! Thank you

**Reply by jugdx** [reply]
Also worked for me! Thank you a lot 🙏🏼

**Reply by restebanez** [reply]
It's worked for me on a MAX M3 with Sonoma 14.3

**Reply by thomasbritton** [reply]
I'm having issues installing `3.0.6` and none of the suggestions in this thread are working for me using rbenv.

It is not giving a very clear error message either which isn't helping! and the log doesn't give any guidance either.

Here is a screenshot:

![Screenshot 2024-01-29 at 17 11 29](https://github.com/rbenv/ruby-build/assets/272787/0fe8b754-0834-4e15-80b9-ee6b71122f13)

**Reply by menti-ashfaq** [reply]
It worked for me as well 😁

**Reply by mislav** [reply]
@thomasbritton From your screenshot, it looks like the `make` step failed, but you didn't share the log file so we are unable to actually see the error message which could indicate the problem. Without a log file, it's little likely that anyone will be able to help you.

**Reply by jiangxiaopeng** [reply]
it worked for me , mac os 14.3.1.

**Reply by umr55766** [reply]
Worked! Thx

**Reply by biancaiz** [reply]
This worked for me too, thank you!
(version 14.4.1)

**Reply by LedZeck** [reply]
It did work for me, thank you!!

but I had to turn off the Rosetta option on iTerm in order to work.

**Reply by baochuquan** [reply]
It works for me!

**Reply by jsperezg** [reply]
not all heroes wear capes

**Reply by Etx17** [reply]
Worked too when i needed to isntall ruby 3.0.0, on Mac0S 14.x 
thanks !

**Comment by diogorlo** [comment]
None of the solutions work for Sonoma 14.0. I'm facing it in CircleCI, GitHub actions and locally.

```
Error running '__rvm_make -j12',
BUILD FAILED (macOS 14.0 on x86_64 using ruby-build 20240116)
```

**Reply by mislav** [reply]
@diogorlo You have not specified which ruby version you were trying to install, what was the exact error output (what you've pasted seems lacking), nor supplied the build log. Without this information, no one can give you any advice about why this might have failed.

However, you typically never need to compile Ruby in GitHub Actions, since you can just [use the `setup-ruby` action](https://github.com/ruby/setup-ruby) to hook up a precompiled Ruby version.

**Reply by diogorlo** [reply]
The version did not matter and the action did not work. The solution was to load chruby through the shell.

**Comment by amkisko** [comment]
I was trying to install ruby 3.0.6 on Sonoma 14.3.1 using rbenv:

```
ruby-build: using readline from homebrew
ruby-build: using libyaml from homebrew
ruby-build: using gmp from homebrew
-> ./configure "--prefix=$HOME/.rbenv/versions/3.0.6" --with-openssl-dir=/usr/local/opt/openssl@1.1 --with-readline-dir=/usr/local/opt/readline --with-libyaml-dir=/usr/local/opt/libyaml --with-gmp-dir=/usr/local/opt/gmp --with-ext=openssl,psych,+ --disable-shared
-> make -j 16

BUILD FAILED (macOS 14.3.1 on x86_64 using ruby-build 20231225)

---

bigdecimal.c:249:5: error: 'maybe_unused' attribute cannot be applied to types
  249 |     ENTER(1);
      |     ^
bigdecimal.c:68:33: note: expanded from macro 'ENTER'
   68 | #define ENTER(n) volatile VALUE RB_UNUSED_VAR(vStack[n]);int iStack=0
      |                                 ^
../.././include/ruby/backward/2/attributes.h:168:28: note: expanded from macro 'RB_UNUSED_VAR'
  168 | #define RB_UNUSED_VAR(x) x RBIMPL_ATTR_MAYBE_UNUSED()
      |                            ^
../.././include/ruby/internal/attr/maybe_unused.h:31:39: note: expanded from macro 'RBIMPL_ATTR_MAYBE_UNUSED'
   31 | # define RBIMPL_ATTR_MAYBE_UNUSED() [[maybe_unused]]
      |                                       ^
bigdecimal.c:372:5: error: 'maybe_unused' attribute cannot be applied to types
  372 |     ENTER(1);
      |     ^
```

Does anybody have suggestions how to fix these?

**Reply by maxjustus** [reply]
I ran into this same issue and after trying a number of incantations I was finally able to make it work by using GCC to compile.

The problem here seems to be that newer versions of clang included in MacOS don't like how the Ruby 3.0 C code uses the `[[maybe_unused]]` attribute. We can sidestep that problem by using a different compiler.

```
brew install gcc
export CC=gcc
rbenv install 3.0.6
```

this succeeds for me and outputs:
```
-> curl -q -fL -o ruby-3.0.6.tar.gz https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.6.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 20.3M  100 20.3M    0     0  10.2M      0  0:00:01  0:00:01 --:--:-- 10.2M
==> Installing ruby-3.0.6...

WARNING: ruby-3.0.6 is nearing its end of life.
It only receives critical security updates, no bug fixes.

ruby-build: using readline from homebrew
ruby-build: using libyaml from homebrew
ruby-build: using gmp from homebrew
-> ./configure "--prefix=$HOME/.rbenv/versions/3.0.6" --with-openssl-dir=/opt/homebrew/opt/openssl@1.1 --enable-shared --with-readline-dir=/opt/h
omebrew/opt/readline --with-libyaml-dir=/opt/homebrew/opt/libyaml --with-gmp-dir=/opt/homebrew/opt/gmp --with-ext=openssl,psych,+
-> make -j 10
-> make install
==> Installed ruby-3.0.6 to /Users/max/.rbenv/versions/3.0.6
```

**Reply by sungsong88** [reply]
> I ran into this same issue and after trying a number of incantations I was finally able to make it work by using GCC to compile.
> 
> The problem here seems to be that newer versions of clang included in MacOS don't like how the Ruby 3.0 C code uses the `[[maybe_unused]]` attribute. We can sidestep that problem by using a different compiler.
> 
> ```
> brew install gcc
> export CC=gcc
> rbenv install 3.0.6
> ```
> 
> this succeeds for me and outputs:
> 
> ```
> -> curl -q -fL -o ruby-3.0.6.tar.gz https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.6.tar.gz
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> 100 20.3M  100 20.3M    0     0  10.2M      0  0:00:01  0:00:01 --:--:-- 10.2M
> ==> Installing ruby-3.0.6...
> 
> WARNING: ruby-3.0.6 is nearing its end of life.
> It only receives critical security updates, no bug fixes.
> 
> ruby-build: using readline from homebrew
> ruby-build: using libyaml from homebrew
> ruby-build: using gmp from homebrew
> -> ./configure "--prefix=$HOME/.rbenv/versions/3.0.6" --with-openssl-dir=/opt/homebrew/opt/openssl@1.1 --enable-shared --with-readline-dir=/opt/h
> omebrew/opt/readline --with-libyaml-dir=/opt/homebrew/opt/libyaml --with-gmp-dir=/opt/homebrew/opt/gmp --with-ext=openssl,psych,+
> -> make -j 10
> -> make install
> ==> Installed ruby-3.0.6 to /Users/max/.rbenv/versions/3.0.6
> ```

I tried that still seeing the same error message `bigdecimal.c:249:5: error: 'maybe_unused' attribute cannot be applied to types` 🥲. macOS 14.6.1 on arm64 using ruby-build 20240917. command I run is 'rbenv install 3.0.7'

**Reply by jcavalieri** [reply]
@sungsong88 did you ever solve this?

**Reply by nicksieger** [reply]
I had to whip up a [patch to fix this](https://gist.github.com/nicksieger/03bf346d3a8b63c5f822993b897da418) so I could build 3.0.x on Sequoia. To install, run:

```
cat ruby-3.0.x.patch | rbenv install --patch 3.0.6
```

**Reply by jcavalieri** [reply]
@nicksieger that works for me with 3.0.7 as well. Thank you!

**Reply by hsbt** [reply]
see https://bugs.ruby-lang.org/issues/20760#note-4

**Reply by ekirilov-enova** [reply]
For other fellow Googlers, if using `ruby-install` (your version may vary):

```
ruby-install ruby 3.0.2 --patch /path/to/ruby-3.0.x.patch
```

and then, when asked about which file to patch, `ext/bigdecimal/bigdecimal.c`.

**Reply by taylorgriffinshipt** [reply]
anyone know how I can get this patch working with rvm

**Reply by nicksieger** [reply]
https://rvm.io/rubies/patching

**Reply by taylorgriffinshipt** [reply]
> https://rvm.io/rubies/patching

yeah I tried that but it is failing, will keep looking 🫡

**Reply by brunopagno** [reply]
For those using `asdf` I was able to apply the patch by doing:

```
RUBY_APPLY_PATCHES=$'path/to/the/ruby-3.0.x.patch' asdf install ruby 3.0.7
```

**Reply by estamos** [reply]
> I had to whip up a [patch to fix this](https://gist.github.com/nicksieger/03bf346d3a8b63c5f822993b897da418) so I could build 3.0.x on Sequoia. To install, run:
> 
> ```
> cat ruby-3.0.x.patch | rbenv install --patch 3.0.6
> ```

Appreciated! 👏
That worked for me for me on ruby 3.0.3. and Sequoia 15.0.1 (24A348) on my Intel MacBook.

`cat ruby-3.0.x.patch | rbenv install --patch 3.0.3`

When running
```
export CC=gcc
rbenv install 3.0.3
```

I was receiving the same erors.

**Comment by parreirat** [comment]
Had the same issue, and fixed it by pointing to my installed `openssl@3`:

`RUBY_CONFIGURE_OPTS=--with-openssl-dir=/opt/homebrew/opt/openssl@3 rbenv install 3.3.4`

**Comment by ernonemacher** [comment]
I'm running into a similar issue but nothing above worked so far.

I recently upgraded my Mac from an Intel to a M3 chip. And I transferred everything using the Mac's Migration Assistant.
I was trying to run `pod install` in a Flutter project.
It was returning an error like `mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e' or 'arm64')`

I reinstalled my ruby, rbenv, ruby-build, libyaml, openssl, cocoapds, psych using brew

Somehow now the cocoapods is using the new Ruby version installed by brew. But apparently, gem is still using the old one (2.6.0). So when I try running `gem anything` returns the `incompatible architecture` error.

I tried installing rbenv and installing any version of ruby to change the system one but then the error bellow happens

```
rbenv install 3.3.4

*** Following extensions are not compiled:
openssl:
	Could not be configured. It will not be installed.
	OpenSSL library directory could not be found in '/opt/homebrew/opt/openssl@3/lib'. You might want to fix this error in one of the following ways.
	  * Recompile OpenSSL by configuring it with --libdir=lib  to specify the OpenSSL library directory.
	  * Recompile Ruby by configuring it with --libdir=<dir> to specify the Ruby library directory.
	  * Compile this openssl gem with --with-openssl-include=<dir> and --with-openssl-lib=<dir> options to specify the OpenSSL include and library directories.

psych:
	Could not be configured. It will not be installed.

BUILD FAILED (macOS 14.4 on arm64 using ruby-build 20240903)
```

**Reply by ernonemacher** [reply]
I am not sure how, but it worked. I reinstalled Homebrew and then ran:

```
brew reinstall ruby
echo 'export PATH="/opt/homebrew/opt/ruby/bin:$PATH"' >> ~/.zshrc
export LDFLAGS="-L/opt/homebrew/opt/ruby/lib"
export CPPFLAGS="-I/opt/homebrew/opt/ruby/include"
export PKG_CONFIG_PATH="/opt/homebrew/opt/ruby/lib/pkgconfig"
```

**Comment by balianbaleno** [comment]
after much struggling, this is what worked for M1 on Sequoia, i did used
`arch -x86_64 rvm install 2.6.10 --with-openssl-dir=/usr/local/opt/openssl@1.1`
to install 2.6.10 needed by bundle/pod...

**Comment by mislav** [comment, marked as answer]
Up-to-date answer that should solve most problems from this thread:

```sh
brew install rbenv/tap/openssl@1.1
curl -fsSL https://github.com/ruby/ruby/commit/1dfe75b0beb7171b8154ff0856d5149be0207724.patch | rbenv install 3.0.7 --patch
```

Or, to apply the same patch with asdf instead of rbenv:

```
RUBY_APPLY_PATCHES="https://github.com/ruby/ruby/commit/1dfe75b0beb7171b8154ff0856d5149be0207724.patch" asdf install ruby 3.0.7
```

### Explanation

- Ruby 3.0 is only compatible with OpenSSL 1.x,  but not with macOS system `/usr/bin/openssl` (actually LibreSSL). The earliest OpenSSL version currently maintained in Homebrew core formula is `openssl@3`, which is too new, and Homebrew core dropped support for `openssl@1.1` a while ago; that's why it's necessary to install `openssl@1.1` from our own Homebrew tap.

- Ruby 3.0 has a known bug in `bigdecimal.c` on macOS; that's why a patch is needed. The bug is fixed in Ruby 3.1, but the patch will never get backported to Ruby 3.0 due to its End-of-Life status.

- Other workarounds from this thread such as setting LDFLAGS/CPPFLAGS, using the `gcc` compiler instead of the default clang, using `arch -x86_64`, etc. are outdated and should never be necessary.

I'm marking this as the new answer because this thread keeps surfacing in web searches.

**Reply by rubybui** [reply]
sadly this did not work with me, still having similar error to this https://gist.github.com/ChunAllen/d700773270e6db5bb59c6fa0f2e1e32e

I tried brew install rbenv/tap/openssl@1.1
curl -fsSL https://github.com/ruby/ruby/commit/1dfe75b0beb7171b8154ff0856d5149be0207724.patch | rbenv install 3.0.2 --patch
and even tried patching in folder + recompile and did not work

PS: Somehow doing all that +   openssl brew link --overwrite openssl@1.1 
 RUBY_CONFIGURE_OPTS="--with-openssl-dir=/opt/homebrew/opt/openssl@1.1" rbenv install 3.0.2
 
 help me compile

**Reply by tuananh131001** [reply]
I'm use mise and this works for me (ruby 3.0.3, macos 15.7.1)
```bash
$ brew link --overwrite openssl@1.1
$ MISE_RUBY_APPLY_PATCHES="https://github.com/ruby/ruby/commit/1dfe75b0beb7171b8154ff0856d5149be0207724.patch" MISE_RUBY_CONFIGURE_OPTS="--with-openssl-dir=/opt/homebrew/opt/openssl@1.1" mise install ruby@3.0.3

```

**Reply by ranahamzadev23** [reply]
Thanks @mislav and @rubybui for your lead. The complete command which helped me fix the issue on macOS2026 is
```
`curl -fsSL https://github.com/ruby/ruby/commit/1dfe75b0beb7171b8154ff0856d5149be0207724.patch | RUBY_CONFIGURE_OPTS="--with-openssl-dir=/usr/local/opt/openssl@1.1" rbenv install 3.0.2 --patch`
```
